sklearn/tree/_classes.py

```diff
@@ -400,13 +407,51 @@ def _fit(
         SPLITTERS = SPARSE_SPLITTERS if issparse(X) else DENSE_SPLITTERS
 
         splitter = self.splitter
+        if self.monotonic_cst is None:
+            monotonic_cst = None
+        else:
+            if self.n_outputs_ > 1:
+                raise ValueError(
+                    "Monotonicity constraints are not supported with multiple outputs."
+                )
+            # Check to correct monotonicity constraint' specification,
+            # by applying element-wise logical conjunction
+            # Note: we do not cast `np.asarray(self.monotonic_cst, dtype=np.int8)`
+            # straight away here so as  to generate error messages for invalid
+            # values using the original values prior to any dtype related conversion.
+            monotonic_cst = np.asarray(self.monotonic_cst)
+            if monotonic_cst.shape[0] != X.shape[1]:
+                raise ValueError(
+                    "monotonic_cst has shape {} but the input data "
+                    "X has {} features.".format(monotonic_cst.shape[0], X.shape[1])
+                )
+            unsatisfied_constraints_conditions = (
+                (monotonic_cst != -1) * (monotonic_cst != 0) * (monotonic_cst != 1)
+            )
+            if np.any(unsatisfied_constraints_conditions):
```

Comment:
As suggested in https://github.com/scikit-learn/scikit-learn/pull/13649/files/75242d9bde4ac9d73820104d1116c992923513e5#r1244833667.

```suggestion
            satisfied_constraints_support = np.all(np.isin(monotonic_cst, (-1, 0, 1)))
            if not satisfied_constraints_support:
```